nomad/node_endpoint.go

```diff
@@ -820,10 +820,50 @@ func (n *Node) UpdateAlloc(args *structs.AllocUpdateRequest, reply *structs.Gene
 		return fmt.Errorf("must update at least one allocation")
 	}
 
+	// Ensure that evals aren't set from client RPCs
+	// We create them here before the raft update
+	if len(args.Evals) != 0 {
+		return fmt.Errorf("evals field must not be set ")
+	}
+
 	// Update modified timestamp for client initiated allocation updates
-	now := time.Now().UTC().UnixNano()
+	now := time.Now()
+	var evals []*structs.Evaluation
+
 	for _, alloc := range args.Alloc {
-		alloc.ModifyTime = now
+		alloc.ModifyTime = now.UTC().UnixNano()
+
+		// Add an evaluation if this is a failed alloc that is eligible for rescheduling
+		if alloc.ClientStatus == structs.AllocClientStatusFailed {
+			ws := memdb.NewWatchSet()
```

Comment:
Look up the allocation here to grab the job/namespace. We don't need to send it over the wire. You can do an in memory lookup. Further this won't cause errors during the upgrade path where there are 0.8 servers and 0.7 clients.